modules/internal/Atomics.chpl

```diff
@@ -405,6 +407,7 @@ module Atomics {
     inline proc const waitFor(val:bool, param order: memoryOrder = memoryOrder.seqCst): void {
       on this {
         while (this.read(order=memoryOrder.relaxed) != val) {
+          chpl_comm_ensure_progress();
```

Comment:
Does AtomicT.waitFor also need this?

Reply:
Yes, I know I made that change but lost it somehow. Good catch.